### ISSUE TITLE
Buffer is emptied after reloading via :checktime

### ISSUE THREAD

**Issue by blueyed** [jamessan/vim-gnupg#49]
I've noticed that when a .gpg buffer has changed outside of Vim (e.g. from another Vim instance with vim-gnupg), the buffer will be empty after `:checktime` and selecting to `(L)oad` the buffer anew.

```
W11: Warning: File "foo.txt.gpg" has changed since editing started                     
See ":help W11" for more info.                                                               
[O]K, (L)oad File:
```

I have some autocommands to trigger this automatically, but it also happens with them disabled and manually calling `:checktime`.

It does not happen always though, but e.g. when adding a new line to the top of  the file.

When this happens the buffer contents looks garbled/encrypted already during the `:checktime` prompt.

I could not trigger it when using `:e` instead to reload the file.

**Comment by jamessan** [comment]
From a quick look at Vim's source, this should be treated just like editing a new file so I don't see off-hand why that would result in an empty buffer.  I could see there being extra content in the buffer since I'm just reading the output of the gpg command into the buffer.

I'll test this out some more later, but wanted to get some initial thoughts down.

**Comment by jamessan** [comment]
Well, I tried a simple reproduction but haven't seen it so far.

    $ cat foo | gpg --armor --encrypt > test.asc
    $ vim test.asc
    <C-z>
    $ (echo; cat foo) | gpg --armor --encrypt > test.asc
    $ fg
    l

The buffer is updated with a blank line at the top every time.

Do you have a reliably reproducible method to encounter this?  Ideally when starting vim as `vim -u NONE -N --noplugins --cmd 'runtime plugin/gnupg.vim'` (or equivalent if vim-gnupg is in a bundle).

**Comment by jamessan** [comment]
Note, I don't see the garbled buffer either.

**Comment by blueyed** [comment]
TLDR: might be fixed in the meantime in Vim / Neovim.

I could just reproduce it using `vim -u NONE -N --noplugins --cmd "set rtp+=$PWD" --cmd "verb runtime plugin/gnupg.vim" ~/foo.txt.gpg`

The difference between `:e` and `:checktime` might already be that the prompt gets displayed,, which shifts the buffer's contents.

I have the impression that it's related to the modeline `vi:set foldmethod=marker foldlevel=0:`.
Leaving out `foldlevel=0` (or using `10`) does not seem to trigger it.

Calling `:e` again on the empty buffer, will trigger vim-gnupg to reload it from the encrypted file.

The bug gets triggered through the `silent 1delete` (https://github.com/jamessan/vim-gnupg/blob/master/plugin/gnupg.vim#L586).  Commenting that will not empty the buffer (but keep the additional line of course).

I can easily reproduce it now:
1. open the file in two Vim instances
2. removing a newline from the top of the file
3. run `:checktime` on the other instance (=> everything OK)
4. add a newline to the top of the file
5. run `:checktime` on the other instance (=> buffer is empty)

The problem does not exist in Neovim (0.0.0-alpha+201510190939 (compiled Oct 21 2015)), but in Vim 7.4.812.

But after upgrading Vim (to 7.4.900) I cannot reproduce it anymore, not even when I've tried to bisect it - so rebuilding seems to have fixed it maybe?!

FWIW, it's still an issue with Vim 7.4.683 from the Ubuntu daily-ppa repo (which appears to be broken / not updating anymore; https://code.launchpad.net/~pkg-vim/+recipe/vim-daily - I've contacted https://launchpad.net/~dobey about changing/adjusting Vim's master repo source).

FWIW, it also happened with 3c01375 (which I've used accidentally in between).

**Comment by jamessan** [comment]
Even with your steps and ensuring that the entire buffer is folded due to modelines, I can't reproduce it. :(

Given that the problem seems to have gone away on your end, I'm inclined to close this.

**Comment by blueyed** [comment]
Yeah, let us close it for now.
